src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
-                //System.out.println("using test ship");
-                if(testShip.covers(playerShipPoints.get(i)))
-                    return "Placement overlaps another ship";
-            }
-            Coordinate[] myPoints = new Coordinate[size];
-            for(int i = 0; i < size; i++){
-                Coordinate toAdd = new Coordinate(Across + i, Down);
-                playerShipPoints.add(toAdd);
-                myPoints[i] = toAdd;
-            }
-            getShip(shipName).setLocation(start, end);
-            if(shipName.equals("aircraftCarrier") || shipName.equals("battleship") || shipName.equals("submarine")){
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerMilitaryShips[militaryPlaceIndex] = (MilitaryShip) getShip(shipName);
-                militaryPlaceIndex++;
-            }
-            else{
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerCivilianShips[civilianPlaceIndex] = (CivilianShip) getShip(shipName);
-                civilianPlaceIndex++;
-            }
-        } else { //horizantal
-            if((Down + size -1) > BOARD_SIZE)
-                return "Ship Placement out of bounds";
+        }
+        else if(orientation.equals("horizontal")) {
             endDown = Down + size -1;
             endAcross = Across;
-            if(endDown > BOARD_SIZE )
-                return "Ship placement out of bounds";
-            Coordinate start = new Coordinate(Across, Down);
-            Coordinate end = new Coordinate(endAcross, endDown);
-            testShip = new MilitaryShip("test", size, start, end, false);
-            for(int i = 0; i < playerShipPoints.size(); i++){
-                if(testShip.covers(playerShipPoints.get(i)))
-                    return "Placement overlaps another ship";
-            }
-            Coordinate[] myPoints = new Coordinate[size];
-            for(int i = 0; i < size; i++){
-                Coordinate toAdd = new Coordinate( Across, Down + i);
-                playerShipPoints.add(toAdd);
-                myPoints[i] = toAdd;
-            }
-            if(shipName.equalsIgnoreCase("aircraftCarrier") || shipName.equalsIgnoreCase("battleship") || shipName.equalsIgnoreCase("submarine")){
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerMilitaryShips[militaryPlaceIndex] = (MilitaryShip) getShip(shipName);
-                militaryPlaceIndex++;
-            }
-            else{
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerCivilianShips[civilianPlaceIndex] = (CivilianShip) getShip(shipName);
-                civilianPlaceIndex++;
-            }
         }
 
```

Comment:
You've refactored a lot here. This section was especially bad, and the changes you have made a definitely paramount to cleaning up our code.